commands/history/inspect.go

```diff
@@ -92,7 +94,135 @@ func runInspect(ctx context.Context, dockerCli command.Cli, opts inspectOptions)
 	}
 	st, _ := ls.ReadRef(rec.node.Builder, rec.node.Name, rec.Ref)
 
-	tw := tabwriter.NewWriter(dockerCli.Out(), 1, 8, 1, '\t', 0)
+	var sg *localstate.StateGroup
+	if st != nil && st.GroupRef != "" {
+		sg, _ = ls.ReadGroup(st.GroupRef)
+	}
+
+	switch opts.format {
+	case formatter.JSONFormatKey:
+		return inspectPrintJSON(ctx, rec, st, sg, dockerCli.Out())
+	case formatter.RawFormatKey:
+		return inspectPrintRaw(ctx, rec, st, dockerCli.Out())
+	default:
+		return errors.Errorf("unsupported format %q", opts.format)
+	}
+}
+
+func inspectPrintJSON(ctx context.Context, rec *historyRecord, ls *localstate.State, lsg *localstate.StateGroup, w io.Writer) error {
+	type buildError struct {
+		Name    string `json:"name,omitempty"`
+		Sources string `json:"sources,omitempty"`
+		Logs    string `json:"logs,omitempty"`
+	}
+
+	out := struct {
+		Record          *controlapi.BuildHistoryRecord `json:"record"`
+		LocalState      *localstate.State              `json:"localState,omitempty"`
+		LocalStateGroup *localstate.StateGroup         `json:"localStateGroup,omitempty"`
+		Name            string                         `json:"name,omitempty"`
+		DefaultPlatform string                         `json:"defaultPlatform,omitempty"`
+		Status          string                         `json:"status,omitempty"`
+		Duration        time.Duration                  `json:"duration,omitempty"`
+		Error           *buildError                    `json:"error,omitempty"`
+	}{
+		Record:          rec.BuildHistoryRecord,
+		Name:            buildName(rec.FrontendAttrs, ls),
+		LocalState:      ls,
+		LocalStateGroup: lsg,
+		Status:          "completed",
+	}
+
+	if rec.CompletedAt != nil {
+		out.Duration = rec.CompletedAt.AsTime().Sub(rec.CreatedAt.AsTime())
+	} else {
+		out.Duration = rec.currentTimestamp.Sub(rec.CreatedAt.AsTime())
+		out.Status = "running"
+	}
+
+	c, err := rec.node.Driver.Client(ctx)
+	if err != nil {
+		return err
+	}
+
+	workers, err := c.ListWorkers(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to list workers")
+	}
+workers0:
+	for _, w := range workers {
+		for _, p := range w.Platforms {
+			out.DefaultPlatform = platforms.FormatAll(platforms.Normalize(p))
```

Comment:
This is not part of the history record. If needed then this can be set into platform key if one is unset.